src/app/api/user/route.ts

```diff
@@ -0,0 +1,30 @@
+import { NextResponse, NextRequest } from "next/server";
+import connectDatabase from "@/lib/database";
+import UserSignee from "@/models/user-model";
+
+export async function POST(request: NextRequest) {
+
+    const { signeeWalletAddress } = await request.json();
+
+    await connectDatabase();
+
+    try {
+        const existingUser = await UserSignee.findOne({ signeeWalletAddress });
+        if (existingUser) {
+            return new Response(null, {
+                status: 204,
+            })
+        }
+        const newUser = new UserSignee({
+            signeeWalletAddress
+        });
+
+        await newUser.save();
+        return NextResponse.json({ message: "User created successfully" }, { status: 201 }); // Use 201 Created
+
+    } catch (error) {
+        console.error("api/user server Error: ", error);
+        return NextResponse.json({ error: "Something went wrong" }, { status: 500 });
```

Comment:
We always return the original error (is a security issue but it makes the app much usable and user friendly)

Reply:
resolved